src/apps/Permissions/Permissions.js

```diff
@@ -65,244 +35,177 @@ class Permissions extends React.Component {
-                        onOpenEntity={this.handleOpenEntity}
-                      />
-                    )}
-                  </Screen>
-
-                  <Screen position={1} animate={animateScreens}>
-                    {['app', 'entity'].includes(location.screen) && (
-                      <React.Fragment>
-                        {location.screen === 'app' && (
-                          <AppPermissions
-                            app={location.app}
-                            loading={appsLoading}
-                            address={location.address}
-                            onManageRole={this.handleManageRole}
-                          />
-                        )}
-                        {location.screen === 'entity' && (
-                          <EntityPermissions
-                            title="Permissions granted to this entity"
-                            loading={appsLoading || permissionsLoading}
-                            address={location.address}
-                          />
-                        )}
-                      </React.Fragment>
-                    )}
-                  </Screen>
-                </Wrap>
-              </AppLayout>
-
-              <AssignPermissionPanel
-                apps={apps}
-                opened={showAssignPermissionPanel}
-                onClose={this.closeAssignPermissionPanel}
-                wrapper={wrapper}
-              />
-
-              <ManageRolePanel
-                app={location.app}
-                apps={apps}
-                opened={managedRole !== null}
-                onClose={this.closeManageRolePanel}
-                role={managedRole}
+  }, [apps, params, onParamsRequest])
+
+  const location = getLocation(params, apps)
+
+  const managedRole =
+    location.screen === 'app' &&
+    location.app &&
+    location.secondaryScreen === 'role'
+      ? resolveRole(location.app.proxyAddress, location.secondaryData)
+      : null
+
+  return (
+    <React.Fragment>
+      <div
+        css={`
+          // This element is only used to reset
+          // the view scroll using scrollIntoView()
+          position: absolute;
+          top: 0;
+          left: 0;
+          width: 1px;
+          height: 1px;
+        `}
+        ref={scrollTopElement}
+      />
+
+      <Header
+        primary={
+          location.screen === 'app' && location.app ? (
+            <Header.Title>
+              {`${location.app.name} permissions`}
+              <LocalIdentityBadge
+                entity={location.app.proxyAddress}
+                shorten
+                css={`
+                  margin-left: ${2 * GU}px;
+                `}
               />
-            </React.Fragment>
+            </Header.Title>
+          ) : (
+            'Permissions'
           )
-        }}
-      </PermissionsConsumer>
-    )
-  }
+        }
+        secondary={
+          <Button
+            mode="strong"
+            onClick={createPermission}
```

Comment:
Would it make sense to have some populated data when the user clicks on this if they've already selected an app?
Eg: 
![image](https://user-images.githubusercontent.com/3072458/63256121-3b1a8000-c277-11e9-8257-7010dbb01c16.png)

Reply:
Hmmm, if I choose an app, close the panel and come back the selected option persists.

Reply:
and because of that the second dropdown (action) does not appear, and the user cannot click on add permission and is not given any feedback:

![image](https://user-images.githubusercontent.com/3072458/63256402-c85dd480-c277-11e9-83aa-f72524dc66d5.png)

Reply:
> Would it make sense to have some populated data when the user clicks on this if they've already selected an app?

I think it depends if we want the main action to be contextual or not /cc @dizzypaty 

>  Hmmm, if I choose an app, close the panel and come back the selected option persists.

It’s not the case here, the state should reset as soon as the panel closes. Did you reopen the panel from the panel using the “New permission” button from an app screen?